data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart

```diff
@@ -120,7 +120,9 @@ class SharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataSource
 
             developer.log('Cannot cast workgroup node = $workgroupNode');
             return null;
-          }).toList();
+          })
+          .where((node) => node != null)
```

Comment:
hope to see you create an extensions `mapNotNull` in the future (of course, not in this PR)